### PR TITLE
RE-rebase of ask a question form

### LINKED ISSUE #3329: Create the Ask A Question - emails to reference queue
### User story
part of https://github.com/pulibrary/orangelight/issues/3272

### Concrete example
example from findingaids: 
Screen Shot 2023-01-03 at 12 13 15 PM

### Implementation notes, if any
Send email to the Reference queue.

### PR DESCRIPTION
**Pull request by maxkadel**
- This form works but has the exact same flash bug as the email form, which has been pulled into a separate ticket

- ~Still need to work on~
  - [x] Double check params / allowed params 
  - [x] Correct title via the form & controller (may need to pass in from the document)
  - [x] Modal form styling
  - [x] Message width
  - [x] Modal closing after mailing
  - [ ] ~Page displaying success flash after mailing~ [this has been pulled out into its own ticket - see #3365]
  - [x] Email formatting (line breaks)

Connected to https://github.com/pulibrary/princeton_ansible/pull/3644
Closes #3329 
 
Co-authored-by: Jane Sandberg <sandbergja@users.noreply.github.com>
Co-authored-by: Ryan Laddusaw <rladdusaw@users.noreply.github.com>

### CODE REVIEW

File: app/controllers/contact_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+class ContactController < ApplicationController
+  def question
+    @form = AskAQuestionForm.new(question_params)
+    if @form.valid? && @form.submit
+      flash[:success] = 'Your question has been submitted'
+
+      return render "question_success" if request.xhr?
```
Comment (sandbergja) [inline app/controllers/contact_controller.rb:8]: Are the xhr and js responses necessary?

File: app/controllers/feedback_controller.rb [hunk 2]
```diff
@@ -22,6 +23,13 @@ def create
     end
   end
 
+  def ask_a_question
+    respond_to do |format|
+      format.html
+      format.js
```
Comment (sandbergja) [inline app/controllers/feedback_controller.rb:29]: Same question. :-)

File: config/locales/blacklight.en.yml [hunk 3]
```diff
@@ -79,6 +79,9 @@ en:
       barcode_header: ''
       barcode_subheader: ''
       barcode_help: 'If you cannot log in, please email <a href="mailto:fstcirc@princeton.edu">fstcirc@princeton.edu</a>.'
+    ask_a_question:
+      form:
+        title: 'Ask a Question'
```
Comment (sandbergja) [inline config/locales/blacklight.en.yml:84]: Thanks for making this translatable!

**Review by sandbergja — COMMENTED** [review]
Thanks, @maxkadel !  Just one question about the controller responses.

File: app/controllers/contact_controller.rb (hunk 1 above)
Comment (maxkadel) [reply app/controllers/contact_controller.rb]: As far as I can tell, no (this was stolen code), so I've taken them out

**Review by christinach — APPROVED** [review]
Thanks @maxkadel !

**Review by sandbergja — APPROVED** [review]
🦄

**Pull request merged**